File: test/aarch64/scripts/basic_test_pie.py
#
# This file is part of libdebug Python library (https://github.com/libdebug/libdebug).
# Copyright (c) 2024 Roberto Alessandro Bertolini, Gabriele Digregorio. All rights reserved.
# Licensed under the MIT license. See LICENSE file in the project root for details.
#

import unittest

from libdebug import debugger

class BasicTestPie(unittest.TestCase):
    def test_basic(self):
        d = debugger("binaries/basic_test_pie")

        d.run()
        bp = d.breakpoint(0x964, file="binary")
        d.cont()

        assert bp.address == d.regs.pc
        assert d.regs.x0 == 0x4444333322221111

        d.cont()
        d.kill()
        d.terminate()

File: test/common/srcs/basic_test_pie.c
//
// This file is part of libdebug Python library (https://github.com/libdebug/libdebug).
// Copyright (c) 2023-2024 Roberto Alessandro Bertolini. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for details.
//

#include <stdio.h>
#include <stdlib.h>
#include <stdint.h>

uint64_t register_test(uint64_t number)
{
    return number;
}

int main()
{
    uint64_t value;

    value = 0xaabbccdd11223344;

    value = register_test(value);

    printf("Basic test pie: %lx\n", value);

    return 0;
}
